internal/storagetools/check.go

```diff
@@ -0,0 +1,66 @@
+package storagetools
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+
+	"github.com/wal-g/tracelog"
+)
+
+func HandleCheckRead(folder storage.Folder, filenames []string) {
+	_, _, err := folder.ListFolder()
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("failed to list the storage: %v", err)
+	}
+	missing := make([]string, 0)
+	for _, name := range filenames {
+		fdir, fname := path.Split(name)
+		sf := folder
+		if fdir != "" && fdir != "/" {
+			sf = folder.GetSubFolder(fdir)
+		}
+		ok, err := sf.Exists(fname)
```

Comment:
I guess that the regular `sf.Exists` will do the job without the need for subfolder check.